fiftyone/core/collections.py

```diff
@@ -646,55 +646,58 @@ def _sync_samples_last_modified_at(self):
     def _sync_dataset_last_modified_at(self):
         dataset = self._root_dataset
         curr_lma = dataset.last_modified_at
-        lma = self._get_last_modified_at()
+        lma = self._max("last_modified_at")
 
         if lma is not None and (curr_lma is None or lma > curr_lma):
             dataset._doc.last_modified_at = lma
             dataset._doc.save(virtual=True)
 
-    def _get_last_modified_at(self, frames=False):
-        if frames and not self._contains_videos(any_slice=True):
+    def _min(self, path):
+        return self._get_extremum(path, 1)
+
+    def _max(self, path):
+        return self._get_extremum(path, -1)
+
+    def _get_extremum(self, path, order):
```

Comment:
This exists in addition to `min()` and `max()` for two reasons:
1. `$sort + $limit 1` is more efficient than `$group _id: None` when the field is indexed
2. When `path` is a frame-level field, these methods are optimized to directly aggregated on the frames collection, which is something that the `Aggregation` classes do not yet support. In other words, `dataset._max("frames.last_modified_at")` is more performant than `dataset.max("frames.last_modified_at")`

Reply:
Good call, just added 👍